homeassistant/components/hausbus/gateway.py

```diff
@@ -0,0 +1,185 @@
+        """Initialize the system."""
+        self.hass = hass
+        self.config_entry = config_entry
+        self.bridge_id = "1"
+        self.devices: dict[str, HausbusDevice] = {}
+        self.channels: dict[str, dict[tuple[str, str], HausbusEntity]] = {}
+        self.home_server = HomeServer()
+        self.home_server.addBusEventListener(self)
+        self._new_channel_listeners: dict[
+            str, Callable[[HausbusEntity], Coroutine[Any, Any, None]]
+        ] = {}
+
+    def add_device(self, device_id: str, module: ModuleId) -> None:
+        """Add a new Haus-Bus Device to this gateways devices list."""
+        device = HausbusDevice(
+            self.bridge_id,
+            device_id,
+            module.getFirmwareId().getTemplateId()
+            + " "
+            + str(module.getMajorRelease())
+            + "."
+            + str(module.getMinorRelease()),
+            module.getName(),
+            module.getFirmwareId(),
+        )
+        if device_id not in self.devices:
+            self.devices[device_id] = device
+        if device_id not in self.channels:
+            self.channels[device_id] = {}
+
+    def get_device(self, object_id: ObjectId) -> HausbusDevice | None:
+        """Get the device referenced by ObjectId from the devices list."""
+        return self.devices.get(str(object_id.getDeviceId()), None)
+
+    def get_channel_list(
+        self, object_id: ObjectId
+    ) -> dict[tuple[str, str], HausbusEntity] | None:
+        """Get the channel list of a device referenced by ObjectId."""
+        return self.channels.get(str(object_id.getDeviceId()), None)
+
+    def get_channel_id(self, object_id: ObjectId) -> tuple[str, str]:
+        """Get the channel identifier from an ObjectId."""
+        return (str(object_id.getClassId()), str(object_id.getInstanceId()))
+
+    def get_channel(self, object_id: ObjectId) -> HausbusEntity | None:
+        """Get channel from channel list."""
+        channels = self.get_channel_list(object_id)
+        if channels is not None:
+            channel_id = self.get_channel_id(object_id)
+            return channels.get(channel_id, None)
+        return None
+
+    def create_light_entity(
+        self, device: HausbusDevice, instance: ABusFeature, object_id: ObjectId
+    ) -> HausbusLight | None:
+        """Create a light entity according to the type of instance."""
+        if isinstance(instance, Dimmer):
+            return HausbusDimmerLight(
+                object_id.getInstanceId(),
+                device,
+                instance,
+            )
+        if isinstance(instance, Led):
+            return HausbusLedLight(
+                object_id.getInstanceId(),
+                device,
+                instance,
+            )
+        if isinstance(instance, RGBDimmer):
+            return HausbusRGBDimmerLight(
+                object_id.getInstanceId(),
+                device,
+                instance,
+            )
+        return None
```

Comment:
Should this raise instead if the type of light is unsupported?

Reply:
Are there any benefits? I think it would be handled the same way. The exception would also just be ignored.

Reply:
No, there's no special benefit.

Reply:
If there is no benefit, I would like to keep it as is.